azkaban-common/src/test/java/azkaban/user/FileWatcherTest.java

```diff
@@ -26,37 +25,55 @@
 import java.nio.file.Files;
 import java.nio.file.Path;
 import java.nio.file.Paths;
+import java.nio.file.StandardCopyOption;
 import java.nio.file.WatchEvent;
 import java.nio.file.WatchKey;
 import java.util.ArrayList;
 import java.util.List;
 import org.junit.After;
 import org.junit.Before;
+import org.junit.Rule;
 import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
 
 public class FileWatcherTest {
 
-  private final Path PATH = setPath();
+  @Rule
+  public final TemporaryFolder temporaryFolder = new TemporaryFolder();
+
+  private Path PATH;
   private FileWatcher fileWatcher;
 
-  private Path setPath() {
+  private Path setPath() throws IOException {
     final URL configURL = Resources.getResource("test-conf/azkaban-users-test1.xml");
-    final String origpath = configURL.getPath();
-    // Generate a path for test file.
-    final String path = origpath.replace("test1", "file_watcher");
-    return Paths.get(path);
+    final String origPathStr = configURL.getPath();
+
+    // Create a new directory and copy the file in it.
+    final Path workDir = temporaryFolder.newFolder().toPath();
+    // Copy the file to keep original file unmodified
+    final String path = workDir.toString() + "/azkaban-users-test1.xml";
+    final Path filePath = Paths.get(path);
+    Files.copy(Paths.get(origPathStr), filePath, StandardCopyOption.REPLACE_EXISTING);
+    return filePath;
   }
 
   @Before
   public void setUp() throws Exception {
     this.fileWatcher = new FileWatcher(SensitivityWatchEventModifier.HIGH);
+    PATH = setPath();
   }
 
   @After
   public void tearDown() throws Exception {
     if (this.fileWatcher != null) {
       this.fileWatcher.close();
     }
+
+    // Cleanup directories
+    final Path workDir = PATH.getParent();
+    if (Files.exists(workDir)) {
+      Files.deleteIfExists(PATH);
```

Comment:
Does `TemporaryFolder` not take care of cleanup automagically? Is this not one of the reasons to use it?